src/main/groovy/org/spongepowered/gradle/ore/OreDeployTask.groovy

```diff
@@ -0,0 +1,156 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.gradle.ore
+
+import groovy.json.JsonSlurper
+import org.apache.http.HttpEntity
+import org.apache.http.entity.mime.MultipartEntityBuilder
+import org.apache.http.entity.mime.content.FileBody
+import org.apache.http.entity.mime.content.StringBody
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.InvalidUserDataException
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.PublishArtifact
+import org.gradle.api.logging.Logger
+import org.gradle.api.tasks.TaskAction
+import org.gradle.plugins.signing.Signature
+
+import javax.net.ssl.SSLHandshakeException
+
+class OreDeployTask extends DefaultTask {
+
+    String instanceUrl = 'https://ore.spongepowered.org'
+    boolean recommended = true
+    String channel
+    String apiKey
+
+    Configuration deploy = project.configurations.archives
+
+    @TaskAction
+    void run() {
+        Logger logger = project.logger
+        def artifacts = deploy.allArtifacts
+        PublishArtifact plugin = artifacts.find { a -> (a.type == "jar") }
+        PublishArtifact sig = artifacts.find { a -> (a instanceof Signature) }
+        if (plugin == null) {
+            throw new InvalidUserDataException("Plugin file not found.")
+        }
+        if (sig == null) {
+            throw new InvalidUserDataException("Signature file not found.")
+        }
+
+        logger.quiet('Publishing ' + plugin.name + ' to ' + instanceUrl + '.')
+        logger.quiet('  Recommended: ' + recommended)
+        logger.quiet('  Channel: ' + channel)
+
+        if (instanceUrl.endsWith("/")) {
+            instanceUrl = instanceUrl.substring(0, instanceUrl.length() - 1)
+        }
+        URL projectUrl
+        try {
+            String pluginId = project.sponge.plugin.id
+            String str = instanceUrl + '/api/projects/' + pluginId + '/versions/' + project.version
+            projectUrl = new URL(str)
+        } catch (MalformedURLException e) {
+            throw new InvalidUserDataException("Invalid project URL", e)
+        }
+
+        if (apiKey == null) {
+            apiKey = project.property('oreDeploy.apiKey')
+        }
+        if (channel == null) {
+            throw new InvalidUserDataException("Missing channel name.")
+        }
+        HttpEntity requestEntity = MultipartEntityBuilder.create()
```

Comment:
You use the API from Apache's http client here, but send the request using Java's API. Since the full http client is downloaded for this anyway, would it be nicer to use Apache's http client entirely for sending the request?